transactions/btc.ts

```diff
@@ -659,125 +659,122 @@ export async function signOrdinalSendTransaction(
-        fee: modifiedFee,
-        selectedFeeRate,
-      } = await getFee(
-        filteredUnspentOutputs,
-        selectedUnspentOutputs,
-        sumSelectedOutputs,
-        satsToSend,
-        recipients,
-        feeRate,
-        changeAddress,
-        network,
-        ordinalUtxo,
-      );
+  const recipients = [
+    {
+      address: recipientAddress,
+      amountSats: new BigNumber(ordinalUtxo.value),
+    },
+  ];
 
-      calculatedFee = modifiedFee;
-      selectedUnspentOutputs = newSelectedUnspentOutputs;
-      satsToSend = satsToSend.plus(modifiedFee);
-      feePerVByte = selectedFeeRate as BigNumber;
-    }
+  const changeAddress = btcAddress;
 
-    const tx = createOrdinalTransaction(
-      privateKey,
-      ordinalUtxoInPaymentAddress ? '' : taprootPrivateKey,
+  // Calculate transaction fee
+  let calculatedFee: BigNumber = new BigNumber(0);
+  if (!fee) {
+    const {
+      newSelectedUnspentOutputs,
+      fee: modifiedFee,
+      selectedFeeRate,
+    } = await getFee(
+      filteredUnspentOutputs,
       selectedUnspentOutputs,
+      sumSelectedOutputs,
       satsToSend,
       recipients,
+      feeRate,
       changeAddress,
       network,
+      ordinalUtxo,
     );
 
-    if (!ordinalUtxoInPaymentAddress) {
-      // Sign ordinal input at index 0
-      tx.signIdx(hex.decode(taprootPrivateKey), 0);
+    calculatedFee = modifiedFee;
+    selectedUnspentOutputs = newSelectedUnspentOutputs;
+    satsToSend = satsToSend.plus(modifiedFee);
+    feePerVByte = selectedFeeRate as BigNumber;
+  }
 
-      // Sign remaining inputs
-      for (let index = 1; index < selectedUnspentOutputs.length; index++) {
-        tx.signIdx(hex.decode(privateKey), index);
-      }
-    } else {
-      // Sign all inputs with same private key
-      tx.sign(hex.decode(privateKey));
+  const tx = createOrdinalTransaction(
+    privateKey,
+    ordinalUtxoInPaymentAddress ? '' : taprootPrivateKey,
+    selectedUnspentOutputs,
+    satsToSend,
+    recipients,
+    changeAddress,
+    network,
+  );
+
+  if (!ordinalUtxoInPaymentAddress) {
+    // Sign ordinal input at index 0
+    tx.signIdx(hex.decode(taprootPrivateKey), 0);
+
+    // Sign remaining inputs
+    for (let index = 1; index < selectedUnspentOutputs.length; index++) {
+      tx.signIdx(hex.decode(privateKey), index);
     }
+  } else {
+    // Sign all inputs with same private key
+    tx.sign(hex.decode(privateKey));
+  }
 
-    tx.finalize();
+  tx.finalize();
 
-    const signedBtcTx: SignedBtcTx = {
-      tx,
-      signedTx: tx.hex,
-      fee: fee ?? calculatedFee,
-      feePerVByte,
-      total: satsToSend,
-    };
+  const signedBtcTx: SignedBtcTx = {
+    tx,
+    signedTx: tx.hex,
+    fee: fee ?? calculatedFee,
+    feePerVByte,
+    total: satsToSend,
+  };
 
-    return await Promise.resolve(signedBtcTx);
-  } catch (error) {
-    return Promise.reject(error.toString());
-  }
+  return await Promise.resolve(signedBtcTx);
```

Comment:
Sorry, last one 😅 Could you change this to `return signedBtcTx`? We're doing so many unnecessary awaits and Promise wrapping in this file 😋

Reply:
we would need to do these improvements in all of the btc.ts file. I guess that's for a separate PR. But its a start nonetheless 😄 I'll make this change

Reply:
Thanks :)